From 75c1ca810f9df0c1b34e5690134e2c0880443144 Mon Sep 17 00:00:00 2001
From: "Tao, Jing" <jing.tao@intel.com>
Date: Mon, 17 Nov 2014 11:01:52 +0800
Subject: [PATCH 323/338] ov5693-ecs: change some resolution to adopt fw
 cropping

Below sensor settings need to be changed to fit firmware
enabled cropping feature:
2592x1944->2576x1936
2592x1456->2576x1456

Following issue get resolved:
CTS Verifier: capture 2048x1536/preview 1280x720
GMS Back Capture: 2.1M capture

Change-Id: Idaf9e2dd4b1c1afda19a02cbd5d771723a0ea185
Tracked-On: #H1204625895
Signed-off-by: Tao, Jing <jing.tao@intel.com>
Signed-off-by: Tao Wei <wei.tao@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/28932
---
 .../camera/drivers/media/i2c/ov5693-ecs/ov5693.h   | 68 ++++++++++++++++++++--
 1 file changed, 63 insertions(+), 5 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.h b/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.h
index 2f14b3d..a885e24 100755
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.h
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.h
@@ -784,6 +784,37 @@ static struct ov5693_reg const ov5693_2592x1456_30fps[] = {
 	{OV5693_TOK_TERM, 0, 0}
 };
 
+static struct ov5693_reg const ov5693_2576x1456_30fps[] = {
+	{OV5693_8BIT, 0x3501, 0x7b},
+	{OV5693_8BIT, 0x3502, 0x00},
+	{OV5693_8BIT, 0x3708, 0xe2},
+	{OV5693_8BIT, 0x3709, 0xc3},
+	{OV5693_8BIT, 0x3800, 0x00},
+	{OV5693_8BIT, 0x3801, 0x00},
+	{OV5693_8BIT, 0x3802, 0x00},
+	{OV5693_8BIT, 0x3803, 0xf0},
+	{OV5693_8BIT, 0x3804, 0x0a},
+	{OV5693_8BIT, 0x3805, 0x3f},
+	{OV5693_8BIT, 0x3806, 0x06},
+	{OV5693_8BIT, 0x3807, 0xa4},
+	{OV5693_8BIT, 0x3808, 0x0a},
+	{OV5693_8BIT, 0x3809, 0x10},
+	{OV5693_8BIT, 0x380a, 0x05},
+	{OV5693_8BIT, 0x380b, 0xb0},
+	{OV5693_8BIT, 0x380c, 0x0a},
+	{OV5693_8BIT, 0x380d, 0x80},
+	{OV5693_8BIT, 0x380e, 0x07},
+	{OV5693_8BIT, 0x380f, 0xc0},
+	{OV5693_8BIT, 0x3811, 0x18},
+	{OV5693_8BIT, 0x3813, 0x00},
+	{OV5693_8BIT, 0x3814, 0x11},
+	{OV5693_8BIT, 0x3815, 0x11},
+	{OV5693_8BIT, 0x3820, 0x00},
+	{OV5693_8BIT, 0x3821, 0x1e},
+	{OV5693_8BIT, 0x5002, 0x00},
+	{OV5693_TOK_TERM, 0, 0}
+};
+
 /*
  * 2592x1944 30fps 0.6ms VBlanking 2lane 10Bit
  */
@@ -896,6 +927,33 @@ static struct ov5693_reg const ov5693_736x496_30fps[] = {
 	{OV5693_TOK_TERM, 0, 0}
 };
 
+static struct ov5693_reg const ov5693_2576x1936_30fps[] = {
+	{OV5693_8BIT, 0x3501, 0x7b},
+	{OV5693_8BIT, 0x3502, 0x00},
+	{OV5693_8BIT, 0x3708, 0xe2},
+	{OV5693_8BIT, 0x3709, 0xc3},
+	{OV5693_8BIT, 0x3803, 0x00},
+	{OV5693_8BIT, 0x3806, 0x07},
+	{OV5693_8BIT, 0x3807, 0xa3},
+	{OV5693_8BIT, 0x3808, 0x0a},
+	{OV5693_8BIT, 0x3809, 0x10},
+	{OV5693_8BIT, 0x380a, 0x07},
+	{OV5693_8BIT, 0x380b, 0x90},
+	{OV5693_8BIT, 0x380c, 0x0a},
+	{OV5693_8BIT, 0x380d, 0x80},
+	{OV5693_8BIT, 0x380e, 0x07},
+	{OV5693_8BIT, 0x380f, 0xc0},
+	{OV5693_8BIT, 0x3811, 0x18},
+	{OV5693_8BIT, 0x3813, 0x00},
+	{OV5693_8BIT, 0x3814, 0x11},
+	{OV5693_8BIT, 0x3815, 0x11},
+	{OV5693_8BIT, 0x3820, 0x00},
+	{OV5693_8BIT, 0x3821, 0x1e},
+	{OV5693_8BIT, 0x5002, 0x00},
+	{OV5693_8BIT, 0x0100, 0x01},
+	{OV5693_TOK_TERM, 0, 0}
+};
+
 struct ov5693_resolution ov5693_res_preview[] = {
 	{
 		.desc = "ov5693_1616x1216_30fps",
@@ -913,7 +971,7 @@ struct ov5693_resolution ov5693_res_preview[] = {
 	},
 	{
 		.desc = "ov5693_5M_30fps",
-		.width = 2592,
+		.width = 2576,
 		.height = 1456,
 		.pix_clk_freq = 160,
 		.fps = 30,
@@ -923,12 +981,12 @@ struct ov5693_resolution ov5693_res_preview[] = {
 		.bin_factor_x = 1,
 		.bin_factor_y = 1,
 		.bin_mode = 0,
-		.regs = ov5693_2592x1456_30fps,
+		.regs = ov5693_2576x1456_30fps,
 	},
 	{
 		.desc = "ov5693_5M_30fps",
-		.width = 2592,
-		.height = 1944,
+		.width = 2576,
+		.height = 1936,
 		.pix_clk_freq = 160,
 		.fps = 30,
 		.used = 0,
@@ -937,7 +995,7 @@ struct ov5693_resolution ov5693_res_preview[] = {
 		.bin_factor_x = 1,
 		.bin_factor_y = 1,
 		.bin_mode = 0,
-		.regs = ov5693_2592x1944_30fps,
+		.regs = ov5693_2576x1936_30fps,
 	},
 };
 #define N_RES_PREVIEW (ARRAY_SIZE(ov5693_res_preview))
-- 
1.9.1

